src/components/ViewSpecRegistrationModal.vue

```diff
@@ -319,10 +319,10 @@ export default defineComponent({
     const handleChangedItem = (item) => {
       if (!item) { return }
 
-      const itemAdded = selectedScopes.value.filter(curr => curr.value === item.value)
+      const itemAdded = selectedScopes.value?.filter(curr => curr.value === item.value)
```

Comment:
if you change this to `.includes`, I think it'd be more clear (i.e. it would return `undefined` if not found), so that on line 325 you don't have to check for length.

Just a nit/food-for-thought.